rmm/src/realm/vcpu.rs

```diff
@@ -87,31 +71,33 @@ pub enum State {
     Running = 2,
 }
 
-pub unsafe fn current() -> Option<&'static mut VCPU<crate::realm::context::Context>> {
+pub unsafe fn current() -> Option<&'static mut VCPU> {
     match TPIDR_EL2.get() {
         0 => None,
-        current => Some(&mut *(current as *mut VCPU<crate::realm::context::Context>)),
+        current => Some(&mut *(current as *mut VCPU)),
     }
 }
 
-pub fn create_vcpu(id: usize) -> Result<usize, Error> {
-    let realm = get_realm(id).ok_or(Error::RmiErrorInput)?;
-
-    let page_table = realm.lock().page_table.lock().get_base_address();
-    let vttbr =
-        bits_in_reg(VTTBR_EL2::VMID, id as u64) | bits_in_reg(VTTBR_EL2::BADDR, page_table as u64);
+pub fn create_vcpu(rd: &mut Rd) -> Result<usize, Error> {
+    let page_table = rd.s2_table().lock().get_base_address();
+    let vttbr = bits_in_reg(VTTBR_EL2::VMID, rd.id() as u64)
+        | bits_in_reg(VTTBR_EL2::BADDR, page_table as u64);
 
-    let vcpu = VCPU::new(realm.clone());
+    let vcpu = VCPU::new();
     vcpu.lock().context.sys_regs.vttbr = vttbr;
     timer::init_timer(&mut vcpu.lock());
     gic::init_gic(&mut vcpu.lock());
 
-    realm.lock().vcpus.push(vcpu);
-    let vcpuid = realm.lock().vcpus.len() - 1;
+    rd.vcpus.push(vcpu);
+    let vcpuid = rd.vcpus.len() - 1;
     Ok(vcpuid)
 }
 
 pub fn remove(id: usize) -> Result<(), Error> {
-    RMS.lock().1.remove(&id).ok_or(Error::RmiErrorInput)?;
+    VMID_SET
+        .lock()
+        .remove(&id)
+        .then_some(())
+        .ok_or(Error::RmiErrorInput)?;
     Ok(())
```

Comment:
[minor] I guess it's enough with `ok_or`

```suggestion
        .ok_or(Error::RmiErrorInput)
```

Reply:
Thank you for the suggestion! I've reflected the suggestion :)